scripts/script.js

```diff
@@ -0,0 +1,127 @@
+import { createDOMElement, gettingError, firstCharToUpperCase, responseErrorChecker } from './helpers.js';
+import { apiKey, cityList } from './data.js';
+import { CityInfo } from './city.js';
+
+const elements = {
+    citySelected: document.querySelector('.weather-city-selected'),
+    cityList: document.querySelector('.weather-city-selection'),
+    images: document.querySelectorAll('.weather-image'),
+    temperatures: document.querySelectorAll('.weather-temperature'),
+    currentState: document.querySelector('.weather-current-state'),
+    currentTime: document.querySelector('.weather-current-time'),
+    currentCity: document.querySelector('.weather-current-city'),
+    currentInfo: document.querySelector('.weather-current-info'),
+    direction: document.querySelector('.icon-direction-pointer'),
+    wind: document.querySelector('.weather-wind'),
+    pressure: document.querySelector('.weather-pressure'),
+    humidity: document.querySelector('.weather-humidity'),
+    dew: document.querySelector('.weather-dew'),
+    visibility: document.querySelector('.weather-visibility'),
+    loading: document.querySelector('.is-loading')
+};
+Object.freeze(elements)
+
+function userGeolocation() {
+    navigator.geolocation.getCurrentPosition(
+        (data) => {
+            const lat = data.coords.latitude;
+            const lon = data.coords.longitude;
+            findCityByGeolocation(lat, lon)
+        },
+        (error) => {
+            gettingError(error.message)
+            if (!window.localStorage.getItem("userCity")) {
+                findCityByName(cityList[0])
+            } else {
+                findCityByName(window.localStorage.getItem('userCity'))
+            }
+        }
+    )
+}
+
+function createCityList(list) {
+    for (let i = 0; i < list.length; i++) {
+        const { cityList } = elements
+        const city = createDOMElement('li', 'weather-city-city')
+        city.innerHTML = list[i]
+        city.setAttribute("data-for", list[i])
+        cityList.append(city)
+    }
+}
+
+function toggleList() {
+    const { cityList } = elements
+    cityList.classList.toggle('hidden')
+}
+
+function animation() {
+    const { loading } = elements
+    loading.classList.remove('is-loading')
+}
+
+function selectedCityUpdate() {
+    const { citySelected } = elements
+    citySelected.innerHTML = window.localStorage.getItem('userCity')
+}
+
+async function findCityByName(cityName) {
+    const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${apiKey}&units=metric`)
```

Comment:
I'd recommend you to store URL and strings in config.js file
```javascript
const URL = 'https://api.openweathermap.org/';
const fetchWeatherByCity = (city) => `${URL}data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;
const fetchWeatherByGeolocation = (lat, lon) => `${URL}data/2.5/weather?lat=${lat}&lon=${lon}&appid=${apiKey}&units=metric`;

export {URL, fetchWeatherByCity, fetchWeatherByGeolocation}
```
and import them in your current file
```javascript
import {URL, fetchWeatherByCity, fetchWeatherByGeolocation} from './components/config.js'
// ...
    const response = await fetch(`fetchWeatherByCity(cityName)`)
```

Reply:
Also you should wrap your async functions into try/catch block
```javascript
async function findCityByName(cityName) {
  try {
     const response = await fetch(`fetchWeatherByCity(cityName)`);
     const data = await response.json();
     if (!response.ok) {
         responseErrorChecker(data);
     }
     updater(data);
  } catch (error) {
      console.log(error);
  }
  ```